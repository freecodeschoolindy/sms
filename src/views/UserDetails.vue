<template>
  <div>
    <h1>{{ userType }} Profile</h1>

    <h2>Publicly visible:</h2>

    <base-text-field
      v-for="field in fields"
      v-model="user[field.id]"
      :id="field.id"
      :label="field.label"
      :placeholder="field.placeholder"
      :key="field.id"
    />

  </div>
</template>

<script>
import BaseTextField from '@/components/utilities/BaseTextField.vue';

export default {
  name: 'UserDetails',
  components: {
    BaseTextField
  },
  data: function () {
    return {
      userType: 'Student',
      user: {
        firstName: 'Jane',
        preferredName: 'Janey',
        lastName: 'Doe',
        codepenUsername: 'jdoe',
        githubUsername: 'jdoe'
      },
      fields: [
        {
          id: 'firstName',
          label: 'First Name',
          placeholder: 'Richard'
        },
        {
          id: 'lastName',
          label: 'Last Name',
          placeholder: 'Sanchez'
        },
        {
          id: 'preferredName',
          label: 'Preferred Name',
          placeholder: 'Rick'
        },
        {
          id: 'githubUsername',
          label: 'GitHub Username',
          placeholder: 'rsanchez'
        },
        {
          id: 'codepenUsername',
          label: 'CodePen Username',
          placeholder: 'rsanchez'
        }
        /*
          Public Profile: Visible to others logged in, maybe to those not logged in?
            Certificates/Badges

          Private Profile: Visible to only the user and organizers/admins
            FreeCodeCamp.org progress page url: freecodecamp.org/codecafecentral
            Email
            Discord Name
            Phone
            Attendenance (not editable by students)
            List of labs:
              Link to the lab starter
              input field for a link to their completed version
        */
      ]
    };
  },
  methods: {},
  computed: {}
};
</script>
