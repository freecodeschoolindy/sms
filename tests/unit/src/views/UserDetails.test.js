import UserDetails from '@/views/UserDetails.vue';

import testHelpers from '@@/testHelpers.js';

describe('UserDetails.vue', () => {
  let localVue;
  let store;
  let setup;

  let permissions = ['student'];

  beforeEach(() => {
    setup = testHelpers.setupTestVariables(permissions);
    localVue = setup.localVue;
    store = setup.store;
  });

  test('Renders correctly', async () => {
    const wrapper = await testHelpers.shallowMountWrapper(UserDetails, {
      localVue,
      store
    });

    expect(wrapper)
      .toMatchSnapshot();
  });
});
