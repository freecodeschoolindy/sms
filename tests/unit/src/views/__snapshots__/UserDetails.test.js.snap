// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails.vue Renders correctly 1`] = `
<div>
  <h1>Student Profile</h1>
  <h2>Publicly visible:</h2>
  <base-text-field-stub label="First Name" id="firstName" value="Jane" placeholder="Richard"></base-text-field-stub>
  <base-text-field-stub label="Last Name" id="lastName" value="Doe" placeholder="Sanchez"></base-text-field-stub>
  <base-text-field-stub label="Preferred Name" id="preferredName" value="Janey" placeholder="Rick"></base-text-field-stub>
  <base-text-field-stub label="GitHub Username" id="githubUsername" value="jdoe" placeholder="rsanchez"></base-text-field-stub>
  <base-text-field-stub label="CodePen Username" id="codepenUsername" value="jdoe" placeholder="rsanchez"></base-text-field-stub>
</div>
`;
